tests/baselines/reference/jsFileFunctionOverloads.types

```diff
@@ -0,0 +1,99 @@
+=== tests/cases/compiler/jsFileFunctionOverloads.js ===
+/**
+ * @overload
+ * @param {number} x
+ * @returns {'number'}
+ */
+/**
+ * @overload
+ * @param {string} x
+ * @returns {'string'}
+ */
+/**
+ * @overload
+ * @param {boolean} x
+ * @returns {'boolean'}
+ */
+/**
+ * @param {unknown} x
+ * @returns {string}
+ */
+ function getTypeName(x) {
+>getTypeName : { (x: number): 'number'; (x: string): 'string'; (x: boolean): 'boolean'; }
+>x : unknown
+
+  return typeof x;
+>typeof x : "string" | "number" | "bigint" | "boolean" | "symbol" | "undefined" | "object" | "function"
+>x : unknown
+}
+
+/**
+ * @template T
+ * @param {T} x 
+ * @returns {T}
+ */
+const identity = x => x;
+>identity : <T>(x: T) => T
+>x => x : <T>(x: T) => T
+>x : T
+>x : T
+
+/**
+ * @template T
+ * @template U
+ * @overload

```

Comment:
just want to double-check: the `@template` has to come before `@overload`, right?
If I recall, parseJSDocSignature fails if it encounters `@template`.

Documentation for the tag will have to make that clear, because it is not at all obvious. It might be a good follow-up feature to parse `@template` in parseJSDocSignature before the param list.

Reply:
> just want to double-check: the @template has to come before @overload, right?

Yup, this is indeed the case. It currently works pretty much the same way `@callback` tag does. Frankly, I have already tried modifying `parseJSDocSignature` to include `@template` tags as `typeParameters` (which I think is always empty at the moment), but it seemed like `typeParameters` are generally ignored for `JSDocSignature` in general, and fixing it felt like a little bit too much for this particular PR. So yes, I agree that this could probably be addressed as a separate feature.